code/__HELPERS/path.dm

```diff
@@ -0,0 +1,328 @@
+#define PATH_DIST(A, B) (A.Distance(B))
+/// Enumerator for the starting turf's sources value, so we know when we've hit the beginning when unwinding at the end
+#define PATH_START	-1
+
+/**
+ * This file contains the stuff you need for using JPS (Jump Point Search) pathing, an alternative to A* that skips
+ * over large numbers of uninteresting tiles resulting in much quicker pathfinding solutions.
+ *
+ * Quick notes about current implementation:
+ * * JPS requires and allows for diagonal movement, whereas our A* only allows for cardinal movement
+ * *
+ * *
+ * *
+ */
+
+
+//JPS nodes variables
+/datum/jpsnode
+	/// The turf associated with this node
+	var/turf/tile
+	/// The node we just came from
+	var/datum/jpsnode/prevNode
+	/// The A* node weight (f = number_of_tiles + heuristic)
+	var/f
+	/// The A* node heuristic (a rough estimate of how far we are from the goal)
+	var/h
+	/// How many steps it's taken to get here from the start (currently pulling double duty as steps taken & cost to get here, since all moves incl diagonals cost 1 rn)
+	var/nt
```

Comment:
why such short variable names? yes, they are docced but there's no reason to use single characters?

Reply:
if you can name variables in such a way that one does not need to go to their definition and documentation to know what they're about, that's better

Reply:
Simply holdovers from the old code